vaults/smart-contracts/v3/TokenizedStrategy.md

```diff
@@ -0,0 +1,1763 @@
+ideal scenario where all the conditions are met.
+
+
+```solidity
+function convertToAssets(uint256 shares) external view returns (uint256);
+```
+**Parameters**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`shares`|`uint256`|The amount of the strategies shares.|
+
+**Returns**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`<none>`|`uint256`|. Expected amount of `asset` the shares represents.|
+
+
+### previewDeposit
+
+Allows an on-chain or off-chain user to simulate
+the effects of their deposit at the current block, given
+current on-chain conditions.
+
+*This will round down.*
+
+
+```solidity
+function previewDeposit(uint256 assets) external view returns (uint256);
+```
+**Parameters**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`assets`|`uint256`|The amount of `asset` to deposits.|
+
+**Returns**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`<none>`|`uint256`|. Expected shares that would be issued.|
+
+
+### previewMint
+
+Allows an on-chain or off-chain user to simulate
+the effects of their mint at the current block, given
+current on-chain conditions.
+
+*This is used instead of convertToAssets so that it can
+round up for safer mints.*
+
+
+```solidity
+function previewMint(uint256 shares) external view returns (uint256);
+```
+**Parameters**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`shares`|`uint256`|The amount of shares to mint.|
+
+**Returns**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`<none>`|`uint256`|. The needed amount of `asset` for the mint.|
+
+
+### previewWithdraw
+
+Allows an on-chain or off-chain user to simulate
+the effects of their withdrawal at the current block,
+given current on-chain conditions.
+
+*This is used instead of convertToShares so that it can
+round up for safer withdraws.*
+
+
+```solidity
+function previewWithdraw(uint256 assets) external view returns (uint256);
+```
+**Parameters**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`assets`|`uint256`|The amount of `asset` that would be withdrawn.|
+
+**Returns**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`<none>`|`uint256`|. The amount of shares that would be burnt.|
+
+
+### previewRedeem
+
+Allows an on-chain or off-chain user to simulate
+the effects of their redemption at the current block,
+given current on-chain conditions.
+
+*This will round down.*
+
+
+```solidity
+function previewRedeem(uint256 shares) external view returns (uint256);
+```
+**Parameters**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`shares`|`uint256`|The amount of shares that would be redeemed.|
+
+**Returns**
+
+|Name|Type|Description|
+|----|----|-----------|
+|`<none>`|`uint256`|. The amount of `asset` that would be returned.|
+
+
+### maxDeposit
+
+Total number of underlying assets that can
+be deposited by `_owner` into the strategy, where `owner`
+corresponds to the receiver of a [deposit](/src/TokenizedStrategy.sol/contract.TokenizedStrategy.md#deposit) call.
```

Comment:
Are links like this autogenerated by forge-doc? Curious how they arrive at this pathing.

Reply:
They don't appear in the original file: https://github.com/yearn/tokenized-strategy/blob/cf791a6f2d360e5c33866c9f0de10e83085920e9/src/TokenizedStrategy.sol#L750

Or at least I think that is the original. Does this have something to do with the { } encapsulation in the natspec comment?

Reply:
So it looks like it is trying to point to the deposit function within itself but the path is wrong since i copied it from the original repo into this one.

I do think that is not a bad feature to have if it could work TBH. That is the idea of using the {} in the natspec because it is signaling that it is referring to a function in the contract.